core/src/main/java/cz/xtf/core/namespace/NamespaceManager.java

```diff
@@ -0,0 +1,194 @@
+            log.info("Start deleting namespace: " + openShift.getNamespace());
+            if (!deleted && waitForDeletion) {
+                BooleanSupplier bs = () -> openShift.getProject() == null;
+                new SimpleWaiter(bs, TimeUnit.MINUTES, 2, "Waiting for " + openShift.getNamespace() + " project deletion")
+                        .waitFor();
+            }
+        }
+        return deleted;
+    }
+
+    private static String getNamespaceForTestClass(TestDescriptor testDescriptor) {
+        if (OpenShiftConfig.useNamespacePerTestcase()) {
+            // some test case names can be really long resulting in long namespace names. This can cause issues
+            // with routes which have 64 chars limit for prefix of domain name. In case of route like:
+            // galleon-provisioning-xml-prio-mnovak-galleonprovisioningxmltest.apps.eapqe-024-dryf.eapqe.psi.redhat.com
+            // route prefix is: galleon-provisioning-xml-prio-mnovak-galleonprovisioningxmltest
+            // route suffix is: .apps.eapqe-024-dryf.eapqe.psi.redhat.com
+            if ((XTFConfig.get(OPENSHIFT_NAMESPACE) + "-"
+                    + testDescriptor.getParent().get().getDisplayName().toLowerCase())
+                            .length() > OpenShiftConfig.getNamespaceLengthLimitForUniqueNamespacePerTest()) {
+
+                return XTFConfig.get(OPENSHIFT_NAMESPACE) + "-"
+                        + StringUtils.truncate(DigestUtils.sha256Hex(testDescriptor.getParent().get().getDisplayName()
+                                .toLowerCase()),
+                                OpenShiftConfig.getNamespaceLengthLimitForUniqueNamespacePerTest()
+                                        - XTFConfig.get(OPENSHIFT_NAMESPACE).length());
+            } else {
+                return XTFConfig.get(OPENSHIFT_NAMESPACE) + "-"
+                        + testDescriptor.getParent().get().getDisplayName().toLowerCase(); // namespace must not have upper case letters
+            }
+        } else {
+            return XTFConfig.get(OPENSHIFT_NAMESPACE);
+        }
+    }
+
+    public static void addTestcaseToNamespaceEntryIfAbsent(TestDescriptor testDescriptor) {
+        getTestcaseToNamespaceMap().putIfAbsent(((MethodBasedTestDescriptor) testDescriptor).getTestClass().getName(),
+                getNamespaceForTestClass(testDescriptor));
+    }
+
+    /**
+     * @return Returns default namespace as defined in xtf.openshift.namespace property or namespace for currently running test
+     *         case when:
+     *         -Dxtf.openshift.namespace.per.testcase=true.
+     *         In case when current thread does not have associated test case (for example when initializing Openshift instance
+     *         in static variable or static block) then {@link java.lang.RuntimeException} exception is thrown.
+     */
+    public static String getNamespace() {
+        if (OpenShiftConfig.useNamespacePerTestcase()) {
```

Comment:
Do we need this if here when same if/else construction is in `getNamespaceForTestcase` ?

Reply:
I mean this https://github.com/xtf-cz/xtf/pull/490/files#diff-5b54c74eced1a279848d833e9dec30d13ae93e5f31a60c45a06298d85f4de130R143

Reply:
Oh, but it is different method :) 

`getNamespaceForTestClass` vs. `getNamespaceForTestcase`

Reply:
But you got my point of eliminating `if (OpenShiftConfig.useNamespacePerTestcase()) {`, right? :)

Can't  `NamespaceManager.getNamespaceForTestcase(TestCaseContext.getRunningTestCaseName());` work for `else` codepath as well?